docs/source/markdown/guides/how_to/data/datasets.md

```diff
@@ -0,0 +1,296 @@
+```{eval-rst}
+:orphan:
+```
+
+# Understanding Anomalib Datasets
+
+This guide explains how datasets work in Anomalib, from the base implementation to specific dataset types and how to create your own dataset.
+
+## Base Dataset Structure
+
+Anomalib's dataset system is built on top of PyTorch's `Dataset` class and uses pandas DataFrames to manage dataset samples. The base class `AnomalibDataset` provides the foundation for all dataset implementations.
+
+### Core Components
+
+The dataset consists of three main components:
+
+1. **Samples DataFrame**: The heart of each dataset is a DataFrame containing:
```

Comment:
Not sure if we should mention the dataframe here, given that it may be replaced soon.